app/src/main/java/tttTDD/Java/Board.java

```diff
@@ -1,44 +1,90 @@
 package tttTDD.Java;
 
-import org.checkerframework.common.value.qual.IntRange;
-
-import java.lang.reflect.Array;
+import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 public class Board {
+    private final int DEFAULT_SIZE = 3;
     private final String[] board;
+    private int numberOfSpaces;
+    private int[][] winningSets = {
+            {0, 1, 2},
+            {3, 4, 5},
+            {6, 7, 8},
+            {0, 3, 6},
+            {1, 4, 7},
+            {2, 5, 8},
+            {0, 4, 8},
+            {2, 4, 6},
+    };
 
     public Board() {
-        this.board = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+        this.board = initializeBoard(DEFAULT_SIZE);
     }
 
     public Board(int boardSize) {
-        this.board = new String[boardSize * boardSize];
-        for (int i = 0; i < this.board.length; i++) {
-            this.board[i] = String.valueOf(i + 1);
-        }
+        this.board = initializeBoard(boardSize);
     }
 
-    public String[] getBoard() {
-        return this.board;
+    public String getSpace(int location) {
+        return isLocationValid(location) ? null : this.board[location];
     }
 
-    public String getSpace(int location) {
-        return this.board[location];
+    public int getFirstAvailableMove() {
+        String[] availableMoves = getAvailableMoves(board);
+        return convertToLocation(availableMoves[0]);
     }
 
-    public void updateBoard(int location, String marker) throws Exception {
+    public boolean updateBoard(int location, String marker) {
         if (isSpaceAvailable(location)) {
             this.board[location] = marker;
-        } else {
-            throw new Exception("The space is already occupied");
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean hasWinningSet() {
+        for (int[] set : winningSets) {
+            if (board[set[0]].equals(board[set[1]]) && board[set[1]].equals(board[set[2]])) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
As we discussed in IPM, this method is currently unsafe to consume from any board size greater than 3, because it will confidently state the game was won or lost based on the marks in the first three rows and columns of the grid.

This should be fixed by either restricting the board size, throwing an error, returning something other than true or false, or implementing an algorithm that's scalable to all board sizes. I put these options roughly in order of the level of effort needed.

Reply:
It's currently in the works. I decided to write a validation that generates a dynamic set of winning moves and then compares them against those locations on the board. We'll see how it pans out once the idea is more developed.

Reply:
This was submitted today with a dynamic board and winning set validation.